runtime/compiler/runtime/JITServerAOTDeserializer.cpp

```diff
@@ -640,6 +642,28 @@ JITServerAOTDeserializer::cacheRecord(const WellKnownClassesSerializationRecord
    }
 
 
+bool
+JITServerAOTDeserializer::cacheRecord(const ThunkSerializationRecord *record,
+                                      TR::Compilation *comp, bool &isNew, bool &wasReset)
+   {
+   if (isResetInProgress(wasReset))
+      return false;
+
+   TR_J9SharedCacheVM *fej9 = (TR_J9SharedCacheVM *)(comp->fe());
+   void *thunk = fej9->getJ2IThunk((char *)record->signature(), record->signatureSize(), comp);
+   if (thunk)
+      return true;
+   isNew = true;
+
+   fej9->setJ2IThunk((char *)record->signature(), record->signatureSize(), record->thunkAddress(), comp);
```

Comment:
I was wondering why we don't check the return value of `setJ2IThunk()`, but apparently we fail the compilation by throwing an exception. I hope that the deserialization code can handle sudden interruptions through exceptions.